internal/benchmarks/src/contract-interaction.bench.ts

```diff
@@ -16,55 +22,93 @@ describe('Contract Interaction Benchmarks', () => {
   let callTestContract: CallTestContract;
   let wallet: WalletUnlocked;
   let cleanup: () => void;
-  beforeEach(async () => {
-    const launched = await launchTestNode({
-      contractsConfigs: [{ factory: CounterContractFactory }, { factory: CallTestContractFactory }],
-    });
+  const isDevnet = process.env.DEVNET_WALLET_PVT_KEY !== undefined;
+  const iterations = isDevnet ? 1 : 10;
 
-    cleanup = launched.cleanup;
-    contract = launched.contracts[0];
-    callTestContract = launched.contracts[1];
-    wallet = launched.wallets[0];
-  });
+  const setupTestEnvironment = async () => {
+    if (isDevnet) {
+      const { networkUrl } = DEVNET_CONFIG;
+      const provider = await Provider.create(networkUrl);
+      wallet = new WalletUnlocked(process.env.DEVNET_WALLET_PVT_KEY as string, provider);
+
+      const { waitForResult } = await CounterContractFactory.deploy(wallet);
+      contract = (await waitForResult()).contract;
 
-  afterEach(() => {
-    cleanup();
+      const { waitForResult: waitForResultCallTestContract } =
+        await CallTestContractFactory.deploy(wallet);
+      callTestContract = (await waitForResultCallTestContract()).contract;
+    } else {
+      const launched = await launchTestNode({
+        contractsConfigs: [
+          { factory: CounterContractFactory },
+          { factory: CallTestContractFactory },
+        ],
+      });
+
+      cleanup = launched.cleanup;
+      contract = launched.contracts[0];
+      callTestContract = launched.contracts[1];
+      wallet = launched.wallets[0];
+    }
+  };
+
+  beforeAll(setupTestEnvironment);
+
+  afterAll(() => {
+    if (!isDevnet && cleanup) {
+      cleanup();
+    }
   });
 
-  bench('should successfully execute a contract read function', async () => {
-    const tx = await contract.functions.get_count().call();
+  const runBenchmark = (name: string, benchmarkFn: () => Promise<void>) => {
+    bench(isDevnet ? name : `${name} (x${iterations} times)`, async () => {
+      for (let i = 0; i < iterations; i++) {
+        await benchmarkFn();
+      }
+    });
+  };
```

Comment:
@maschad Why not simply use `vitest` features?

https://vitest.dev/api/#bench
 - `iteration`
 - `time`
 - `warmupIteration`
 - `warmupTime`
 - `setup`
 - `teardown`

> cc @petertonysmith94

Reply:
I assumed that these options had been explored and that was the reason behind the iterations.

I would favour this.

Reply:
So I've done another set of refactors aea62fb1f 

But @arboleya @petertonysmith94 apart from the `iterations` option, where would I use the other options?

The diff now is 

```ts
const runBenchmark = (name: string, benchmarkFn: () => Promise<void>) => {
    bench(isDevnet ? name : `${name} (x${iterations} times)`, async () => {
        await benchmarkFn();
    }, {iterations});
  };
```

Which is not substantial, but I agree preferable. Recall the test name is contingent on whether it's devnet or a local node.